tla/trace2scen.py

```diff
@@ -0,0 +1,73 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the Apache 2.0 License.
+
+import sys
+import json
+import os
+
+def comment(action):
+    return f"# {action['name']} {action['location']['module']}:{action['location']['beginLine']}"
+
+def term(ctx, pre):
+    return str(pre["currentTerm"][ctx['i']])
+
+def noop(ctx, pre, post):
+    return ["# Noop"]
+
+MAP = {
+    "ClientRequest": lambda ctx, pre, post: ["replicate", term(ctx, pre), "42"],
```

Comment:
Is there a reason these return lists, joined on line 58, rather than formatting directly?

eg

```suggestion
    "ClientRequest": lambda ctx, pre, post: f"replicate,{term(ctx,pre)},42",
```

If a trace action ever results in multiple scenario steps, then we'll want a list of results then to be joined later, and then the nested lists will be more confusing.

Reply:
I was mostly trying to keep everything structured until a final flattening step, to facilitate any additional passes we may need to do. But we don't have any, and we could just as easily format directly here.